### PR TITLE
update release instructions for posting

### PR DESCRIPTION
**Pull request by jodygarnett**
Signed-off-by: Jody Garnett <jody.garnett@gmail.com>

### CODE REVIEW

File: doc/en/developer/source/release-guide/index.rst [hunk 1]
```diff
@@ -404,49 +404,10 @@ The following is an example::
...
-    * Default Templates per Namespace
-
-  Along with many other improvements and bug fixes. The entire change log for
-  the 1.7.1 series is available in the issue tracker:
-  
-  https://osgeo-org.atlassian.net/jira/secure/ReleaseNote.jspa?projectId=10000&version=
+For major releases it is worth making a news announcement:
+   
+* Project Steering Comittee members should have website access and can post a news item directly
+* The instructions for `submit news <http://www.osgeo.org/content/news/submit_news.html>`__ ask that email be sent to news_item@osgeo.org
```
Comment (tbarsballe) [inline doc/en/developer/source/release-guide/index.rst]: For simplicity and clarity, just describe a single, accepted approach. As-is, it is unclear what the releaser should do here.

**Comment by aaime** [comment]
Looks good to me, +1

File: doc/en/developer/source/release-guide/index.rst (hunk 1 above)
Comment (jodygarnett) [reply doc/en/developer/source/release-guide/index.rst]: Sadly I cannot since it should change shortly with the new osgeo website.

Comment (tbarsballe) [reply doc/en/developer/source/release-guide/index.rst]: How shortly is shortly? Perhaps just wait on this PR until we actually have an accepted approach?

**Comment by jodygarnett** [comment]
Revised based on feedback, @tbarsballe are you happy with the change as written (less specific instructions).

**Comment by tbarsballe** [comment]
Sure, although lets try and remember to add more specific instructions once they exist.

**Pull request merged**